src/ThreadMonitor.h

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#elif __APPLE__
+#elif __linux__
+#include <unistd.h>
+#endif
+
+#include <fstream>
+#include <vector>
+
+namespace visor {
+
+class ThreadMonitor
+{
+public:
+    static inline double cpu_percentage()
```

Comment:
this is not right yet. I would like to know which approach we should use to get cpu percentage. most of them uses a time delay and calculate the clock difference between this time